src/serlio/materials/ArnoldMaterialNode.h

```diff
@@ -19,24 +19,25 @@
 
 #pragma once
 
-#include "maya/MPxNode.h"
+#include "MaterialNode.h"
 
 class MaterialInfo;
-class MaterialTrafo;
 class MELScriptBuilder;
 
-class ArnoldMaterialNode : public MPxNode {
-
+class ArnoldMaterialNode : public MaterialNode {
 public:
 	static MStatus initialize();
-
 	static MTypeId id;
-	static MObject aInMesh;
-	static MObject aOutMesh;
-
-	MStatus compute(const MPlug& plug, MDataBlock& data) override;
 
-	MPxNode::SchedulingType schedulingType() const noexcept override {
-		return SchedulingType::kGloballySerial;
-	}
+	static MObject mInMesh;
+	static MObject mOutMesh;
+
+private:
+	void declareMaterialStrings(MELScriptBuilder& sb);
+	void appendToMaterialScriptBuilder(MELScriptBuilder& sb, const MaterialInfo& matInfo,
+	                                   const std::wstring& shaderBaseName, const std::wstring& shadingEngineName);
+	std::wstring getBaseName();
+	MObject getInMesh();
+	MObject getOutMesh();
+	std::vector<std::string> getPluginDependencies();

```

Comment:
use `override` and `const` keywords for these virtual functions

Reply:
done